### ISSUE TITLE
Use fixed list of (managed?) funders on project detail page

### ISSUE THREAD

**Issue by nicolasfranck** [DMPbelgium/roadmap#64]
Please complete the following fields as applicable:

**What version of the DMPRoadmap code are you running? (e.g. v2.2.0)**

`3.1.1` (see test.dmponline.be at 2022-12-22) 

**TODO:**

* The funder you can choose on the project detail page should be filled with funders as drawn from https://api.crossref.org/funders.
* That list is very long (32248 items), and should probably be limited to Belgium (and near countries?)
* Use select box or use a popup dialog where can type and search for the funder
* You need to prefill the table `orgs` with organisations with org type "funder", and set the flag `managed` to `true` (to distinguish from user input), and then use that managed list of funders:
  * When I fetch the data from crossref I start from a URI containing a DOI (e.g. https://doi.org/10.13039/100000001), which I have to lookup in table `identifiers` (`SELECT * FROM identifiers WHERE value = 'https://doi.org/10.13039/100000001' AND identifiable_type = 'Org' AND identifier_scheme_id IN (SELECT id FROM identifier_schemes WHERE name = 'fundref'))`)
  * If the identifier does not exist yet: create it
  * If the organisation does not exist yet: create it with the name of the funder in crossref
  * If the organisation does exist: set managed=true and set funder=true. I lookup the organisation by searching on column `name` in mysql, which is an exact search. For some organisations that will work, for other not (e.g. name for Ghent University is different in crossref).
  * This will make a lot of new funders, without templates
  * This will make some existing organisations funder, and therefore give certain organisation a double role as both "research institution" and "funder". So you can see them in both "research organisation" and "funder" on the plan creation wizard. On template level you will see the template as "customizable" (i.e. "a funder template") as soon as the visibility is reset from "organisationally_visible" to "publicly_visible". So existing templates of an organisation will not suddenly show up as funder templates, unless you change that option.

Is it actually desirable to have this sudden increase in managed organisations that are only used once in a while?
Additionally, for some existing organisations, that are also funder in crossref, but are named differently here (e.g. Ghent University), an extra table entry will be made, as the match on name does not work properly.

Is a manual entered list of funders not more desirable? Based on the existing unmanaged list of funders? So fixing those records, adding the fundref identifiers, and then set them to managed?

**Comment by nicolasfranck** [comment]
Some important notes about the current working:

* the "create plan wizard" shows all funders that have published templates. Indirectly this also means that those funders are managed organisations. See https://github.com/DMPbelgium/roadmap/blob/master/app/controllers/plans_controller.rb#L41
* the "project details" page, where you also can select a funder, lists the same funders (so those with published templates). See https://github.com/DMPbelgium/roadmap/blob/master/app/controllers/plans_controller.rb#L179
* so simply adding managed organisations of type funder in the table "orgs" would not influence that list on the project detail page.
* I can of course override [this line](https://github.com/DMPbelgium/roadmap/blob/master/app/views/branded/plans/_project_details.html.erb#L158) and fetch all managed organisations of type funder. That would add other values to the autocomplete, but would still leave the input empty if none of the values matches what we entered. Luckily, luckily, if we do not change the input, the value is NOT overwritten.

So in short: simply keep using this autocomplete field, but draw the autocomplete from all managed funders? 

About the list of funders: once new funders will not be created anymore, we can start "fixing" the old values, or leave that up to the users? Ah right, there is also the "merge organisation" tool, right?

**Comment by nicolasfranck** [comment]
Some problems:

* There are just too many funders available in Crossref.
* How to automatically upgrade existing "funder" organisations if the name does not match?

**Comment by nicolasfranck** [comment]
Some remarks about the funders attached to plans:

* `219` unique values:

```
> funder_ids = Plan.where("funder_id is not null and funder_id <> ''").map(&:funder_id).uniq
=> 219
```
* only `19` of them are `managed`:

```
irb(main):049:0> Org.where(id: funder_ids).select(&:managed?).size
=> 19
```

* only `8` of them are marked as funder:

```
irb(main):050:0> Org.where(id: funder_ids).select(&:funder?).size
=> 8
```

Apparently organisations that were automatically created by use of the "edit project details" page were never marked as funder (see [org_from_params](https://github.com/DMPRoadmap/roadmap/blob/main/app/controllers/concerns/org_selectable.rb#L68) and [place](https://github.com/DMPRoadmap/roadmap/blob/main/app/controllers/plans_controller.rb#L108) where it is used). The same is true though for institutions attached to plans: https://github.com/DMPRoadmap/roadmap/blob/main/app/controllers/plans_controller.rb#L100. But I guess, if organisations are automatically (and unwillingly) created by users when editing plans, these marks ("funder", "institution") are probably seen as opinions.
